### PR TITLE
Implement response::getType()

### PR DESCRIPTION
**Pull request by saeideng**
https://github.com/cakephp/app/pull/557

### CODE REVIEW

File: src/Http/Response.php [hunk 1]
```diff
@@ -1069,6 +1069,16 @@ public function type($contentType = null)
     }
 
     /**
+     * Retruns the current content type.
```
Comment (ADmad) [inline src/Http/Response.php]: Typo "Retruns"

Comment (saeideng) [reply src/Http/Response.php]: solved

File: src/Http/Response.php [hunk 2]
```diff
@@ -1069,6 +1069,16 @@ public function type($contentType = null)
     }
 
     /**
+     * Returns the current content type.
+     *
+     * @return string
+     */
+    public function getType()
```
Comment (markstory) [inline src/Http/Response.php]: Should we update the deprecation notice for `Response::type()`

Comment (saeideng) [reply src/Http/Response.php]: I do it in new PR after merging this 
+
implementing `public function withMimeType(array $mimeTypes)`

Comment (saeideng) [reply src/Http/Response.php]: also https://github.com/cakephp/cakephp/pull/11336#issuecomment-337341305

Comment (markstory) [reply src/Http/Response.php]: If someone wants to use a content type with no short alias, they should give the full content type. I don't know if we need to allow the short type aliases to be modified.

File: tests/TestCase/Http/ResponseTest.php [hunk 3]
```diff
@@ -214,6 +214,19 @@ public function testType()
...
+     * Tests the getType method
+     *
+     * @return void
+     */
+    public function testGetType()
+    {
+        $response = new Response();
+        $this->assertEquals('text/html', $response->getType());
+        $response->type('pdf');
+        $this->assertEquals('application/pdf', $response->getType());
```
Comment (markstory) [inline tests/TestCase/Http/ResponseTest.php]: Could we get a test for withType() and getType() being used together?

File: src/Http/Response.php (hunk 2 above)
Comment (saeideng) [reply src/Http/Response.php]: Anyway we need a method for adding new mimetypes before full deprecating `type()`

Comment (ADmad) [reply src/Http/Response.php]: @saeideng You already can using `withHeader('Content-Type', 'foo/bar')` :)

Comment (saeideng) [reply src/Http/Response.php]: >withHeader('Content-Type', 'foo/bar')

@ADmad 
so we no longer need to new method( `withMimeType`) :)

Comment (ADmad) [reply src/Http/Response.php]: We could add that convenience too if people think it's worth saving typing of `'Content-Type'` :)

**Comment by markstory** [comment]
Thanks @saeideng !

**Pull request merged**